src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -159,32 +160,24 @@ private[health] class HealthCheckActor(
             timestamp = health.lastFailure.getOrElse(Timestamp.now()).toString
           )
         )
-        val killed = killService.killInstances(Seq(instance), KillReason.FailedHealthChecks)
-        Try(Await.ready(killed, 5 seconds)) match {
-          case Success(future) => future.value.get match {
-            case Success(done) =>
-              instanceTracker.countActiveSpecInstances(app.id).onComplete {
-                case Success(activeInstances) => logger.info(s"Done killing $instanceId, $activeInstances remaining")
-                case Failure(throwable) => logger.warn(s"[anti-snowball] Error when counting instance remaining: ${throwable.getMessage}. Since it is just for logging purpose, we don't care that much")
-              }
-            case Failure(t) => logger.error(s"An error has occurred: ${t.getMessage}", t)
-          }
-          case Failure(_) => logger.error(s"Timed out killing instance $instanceId")
-        }
+        // TODO prevent killingInFlight from growing by cleaning killed instances (probably in purgeStatusOfDoneInstances)
+        killingInFlight = killingInFlight + instanceId
+        logger.debug(s"[anti-snowball] killing ${instanceId}, killingInFlight = ${killingInFlight}")
```

Comment:
Are we sure logging a potentially large set is a good idea?

Reply:
No, this was done for debugging purpose, and should be removed :)